deepchecks/vision/checks/model_evaluation/single_dataset_performance.py

```diff
@@ -0,0 +1,169 @@
+        """Initialize the metric for the check, and validate task type is relevant."""
+        self.scorers = get_scorers_dict(context.get_data_by_kind(dataset_kind), self.user_scorers)
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        for scorer in self.scorers.values():
+            scorer.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        results_dict = {}
+        for name, scorer in self.scorers.items():
+            result = scorer.compute()
+            results_dict[name] = [result] if isinstance(result, Number) else result
+        result_df = metric_results_to_df(results_dict, context.get_data_by_kind(dataset_kind))
+        return CheckResult(result_df)
+
+    def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
+        """Return the values of the metrics for the test dataset in {metric: value} format."""
+        # Find metrics that were reduced over the classes and replace the Class Name with None
+        is_reduced_metrics = check_result.value.groupby('Metric')['Class Name'].nunique() == 1
+        reduced_metrics = is_reduced_metrics[is_reduced_metrics].index.to_list()
+        check_result.value.loc[check_result.value.Metric.apply(lambda x: x in reduced_metrics), 'Class Name'] = None
+
+        # Dict keys are format metric_class
+        metric_class = check_result.value.loc[:, ['Metric', 'Class Name']].aggregate(lambda x:
+                                                                                     '_'.join(filter(None, x)), axis=1)
+        output_dict = dict(zip(metric_class, check_result.value['Value']))
+        return output_dict
+
+    def add_condition_greater_than(self, threshold: float, metrics: List[str] = None, class_mode: str = 'all'):
+        """Add condition - the result is greater than the threshold.
+
+        Parameters
+        ----------
+        threshold: float
+            The threshold that the metrics result should be grater than.
+
+        metrics: List[str]
+            The names of the metrics from the check to apply the condition to. If None, runs on all of the metrics that
+            were calculated in the check.
+
+        class_mode: str, default: 'all'
+            The decision rule over the classes, one of 'any', 'all', class name. If 'any', passes if at least one class
+            result is above the threshold, if 'all' passes if all of the class results are above the threshold,
+            class name, passes if the result for this specified class is above the thershold.
+        """
+
+        def condition(check_result, metrics=metrics):
+            if metrics is None:
```

Comment:
```suggestion
            check_result.value['Metric'].unique() if metrics is None else metrics```

Better not to use self or pycharm is angry

Reply:
Better not to use self or pycharm is angry